node_modules/oae-principals/lib/api.user.js

```diff
@@ -194,8 +197,20 @@ var _createUser = function(ctx, user, callback) {
         // Emit an event indicating a user account has been created
         PrincipalsEmitter.emit(PrincipalsConstants.events.CREATED_USER, ctx, user);
 
-        // Return to the caller
-        return callback(null, user);
+        // If the email address hasn't been verified we ask the user to verify it
+        if (email && !user.email) {
+            _sendEmailToken(ctx, user, email, null, function(err) {
+                if (err) {
+                    return callback(err);
+                }
+
+                return callback(null, user);
+            });
+
+        // Otherwise we can return
+        } else {
+            return callback(null, user);
+        }
```

Comment:
IMO, `invokeIfNecessary` is a better way to do this kind of thing, but I'm feeling quite a bit of resistance on it. Is it too verbose? Would a `_` mixin such as `_.oaeIf(<predicate>, <function>, <params...>, <callback>);` make it a bit more agreeable?

Essentially I'm trying to reduce the amount of copying of logic, and make the natural path of the code a bit easier to follow. In this case:

1. We would be avoiding one level of indentation
2. We would only have 1 "happy path" (e.g., error-less) exit to the method (natural path is easier to follow)
3. We would only have to write `return callback(null, user);` once (reduce copying)